syntax = "proto3";

package dbx1000;

service DBx1000Service {
  // for instance
  rpc LockRemote(LockRemoteRequest) returns (LockRemoteReply) {}
  rpc UnLockRemote(UnLockRemoteRequest) returns (UnLockRemoteReply) {}
  rpc InstanceInitDone(InstanceInitDoneRequest) returns (InstanceInitDoneReply) {}
  rpc BufferManagerInitDone(BufferManagerInitDoneRequest) returns (BufferManagerInitDonReply) {}

  rpc GetTestNum(GetTestNumRequest) returns (GetTestNumReply) {}
}
enum RpcLockMode {
  O = 0;
  P = 1;
  S = 2;
  X = 3;
}
message LockRemoteRequest{
  int32 instance_id = 1;
  uint64 page_id = 2;
  uint64 page_version = 3;
  uint64 key = 4;
  uint64 key_version = 5;
  bytes page_buf = 6;
  uint64 count = 7;
}
message LockRemoteReply{
  bool rc = 1;
  bytes page_buf = 2;
  uint64 count = 3;
}

message UnLockRemoteRequest{
  int32 instance_id = 1;
  uint64 page_id = 2;
  uint64 page_version = 3;
  uint64 key = 4;
  uint64 key_version = 5;
  bytes page_buf = 6;
  uint64 count = 7;
}
message UnLockRemoteReply{
  bool rc = 1;
  bytes page_buf = 2;
  uint64 count = 3;
}

message InstanceInitDoneRequest{
  int32 instance_id = 1;
}
message InstanceInitDoneReply{}

message BufferManagerInitDoneRequest{}
message BufferManagerInitDonReply{
  bool init_done = 1;
}
message  GetTestNumRequest{}
message GetTestNumReply{int32 num = 1;}